lib/modal/modal.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+
+import BEM from '../utilities/ts/bem';
+
+import Button from '../button';
+import Grid from '../grid';
+import GridItem from '../grid-item';
+import Heading from '../heading';
+import ModalImage from './modal-image';
+
+interface TButton {
+  label: string;
+  modifiers: string;
+  onClick: (() => void) | string;
+}
+
+interface Props {
+  buttons: TButton[];
+  children: React.ReactNode;
+  className?: string;
+  message: string;
+  style?: React.CSSProperties;
+}
+
+interface State {
+  mounted: boolean;
+}
+
+export default class Modal extends React.Component<Props> {
+  static defaultProps: Partial<Props> = {
+    buttons: [
+      {
+        label: ``,
+        modifiers: ``,
+        onClick: () => {
+          return;
+        }
+      }
+    ]
+  };
+
+  readonly state: Readonly<State> = {
+    mounted: false
+  };
+
+  componentDidMount(): void {
+    this.showModal();
+  }
+
+  handleClick = (): void => {
```

Comment:
Suggest renaming this function to closeModal or something. as What it does is close the modal.